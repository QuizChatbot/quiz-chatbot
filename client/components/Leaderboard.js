import React, { PropTypes, Component } from 'react'
import { firebaseToArray } from '../utils'

export const LeaderboardItem = ({ idx, developer }) => (
  <h2>{idx + 1}. {developer.profile.first_name} {developer.maxSummary.grade} ( score: {developer.maxSummary.score} )</h2>
)

export class Leaderboard extends Component {
  constructor(props, context) {
    super(props, context)
  }

  render() {
    return (
      <div>
        <h2>Leaderboard</h2>
        {!this.props.developers.length ?
          (<div>No Players</div>)
          : this.props.developers.map((developer, idx) =>
            <LeaderboardItem key={idx} idx={idx} developer={developer} />
          )}
      </div>
    )
  }
}

export default Leaderboard
