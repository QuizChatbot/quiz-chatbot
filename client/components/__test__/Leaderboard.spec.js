import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Leaderboard, LeaderboardItem } from '../Leaderboard'

describe('Leaderboard component test', () => {
  const props = {
    developers: [{
      profile: {
        first_name: 'Rungsikorn'
      },
      maxSummary: {
        grade: 'B',
        score: 75
      }
    }]
  }

  it('should render leaderbaord item message', () => {
    const Wrapper = shallow(<LeaderboardItem developer={props.developers[0]} key={0} idx={0} />)
    expect(Wrapper.contains(<h2>1. Rungsikorn B ( score: 75 )</h2>)).toMatchSnapshot()
  })

  it('should render leaderboard component', () => {
    const Wrapper = shallow(<Leaderboard {...props} />)
    expect(Wrapper.contains(<h2>Leaderboard</h2>)).toMatchSnapshot()
    expect(Wrapper.contains(<LeaderboardItem developer={props.developers[0]} key={0} idx={0} />)).toMatchSnapshot()
  })

  it('should render No players message if developers length is 0', () => {
    const Wrapper = shallow(<Leaderboard developers={[]} />)
    expect(Wrapper.contains('No Players')).toMatchSnapshot()
  })
})