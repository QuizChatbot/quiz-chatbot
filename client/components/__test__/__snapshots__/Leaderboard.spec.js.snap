// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard component test should render No players message if developers length is 0 1`] = `true`;

exports[`Leaderboard component test should render leaderbaord item message 1`] = `true`;

exports[`Leaderboard component test should render leaderboard component 1`] = `true`;

exports[`Leaderboard component test should render leaderboard component 2`] = `true`;
